pkg/virt-launcher/virtwrap/device/sriov/hostdev.go

```diff
@@ -210,3 +223,36 @@ func AttachHostDevices(dom deviceAttacher, hostDevices []api.HostDevice) error {
 	}
 	return nil
 }
+
+// FilterOutHostDevicesByAlias given two slices of host-devices, according to Alias.Name,
+// it returns a slice with host-devices that exists on 'hostDevices' and not exists on 'removeHostDevices'.
+func FilterOutHostDevicesByAlias(hostDevices, removeHostDevices []api.HostDevice) []api.HostDevice {
+	hostDevices = filterHostDevice(hostDevices, hostDeviceHasAlias)
+	removeHostDevices = filterHostDevice(removeHostDevices, hostDeviceHasAlias)
+
+	lookupHostDeviceToRemove := make(map[string]struct{}, len(removeHostDevices))
+	for _, hostDev := range removeHostDevices {
+		lookupHostDeviceToRemove[hostDev.Alias.GetName()] = struct{}{}
+	}
+
+	var filteredSlice []api.HostDevice
+	for idx, hostDev := range hostDevices {
+		if _, exists := lookupHostDeviceToRemove[hostDev.Alias.GetName()]; !exists {
+			filteredSlice = append(filteredSlice, hostDevices[idx])
+		}
+	}
+
+	return filteredSlice
+}
+
+func GetSRIOVHostDevicesToAttach(vmi *v1.VirtualMachineInstance, domainSpec api.DomainSpec) ([]api.HostDevice, error) {
```

Comment:
No need to re-specify "SRIOV", we are under the `sriov` package.

Reply:
Done